globals/matchmaking_globals.go

```diff
@@ -12,3 +12,4 @@ type CommonMatchmakeSession struct {
 }
 
 var Sessions map[uint32]*CommonMatchmakeSession
+var CurrentGatheringID uint32
```

Comment:
Not a massive deal but `nex-go` already provides a `Counter` struct which is used internally there for things like client connection IDs and packet sequence IDs, but can also just be reused here